core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -148,10 +148,19 @@ class AdminClient(val time: Time,
       throw new IllegalArgumentException(s"Consumer Group $groupId with protocol type '${metadata.protocolType}' is not a valid consumer group")
 
     Errors.forCode(metadata.errorCode()).maybeThrow()
-    val consumers = metadata.members.map { consumer =>
-      val assignment = ConsumerProtocol.deserializeAssignment(ByteBuffer.wrap(Utils.readBytes(consumer.memberAssignment)))
-      ConsumerSummary(consumer.memberId, consumer.clientId, consumer.clientHost, assignment.partitions.toList)
-    }.toList
+    val consumers = metadata.state match {
+      case "Stable" =>
+        metadata.members.map { consumer =>
+          val assignment = ConsumerProtocol.deserializeAssignment(ByteBuffer.wrap(Utils.readBytes(consumer.memberAssignment)))
+          ConsumerSummary(consumer.memberId, consumer.clientId, consumer.clientHost, assignment.partitions.toList)
+        }.toList
+      case "PreparingRebalance" | "AwaitingSync" =>
```

Comment:
The current code, when a rebalance is in progress returns this:
```
Error: Consumer group 'group1' is rebalancing.
```

With the change you suggested, the above message would no longer be returned. Instead consumers in the group are listed with no assignment info. E.g.,
```
TOPIC                          PARTITION  CURRENT-OFFSET  LOG-END-OFFSET  LAG        CONSUMER-ID                                       HOST                           CLIENT-ID
-                              -          -               -               -          consumer-1-46cfa20e-229b-40eb-bff0-49403c3156c1   /127.0.0.1                     consumer-1
-                              -          -               -               -          consumer-1-c3862300-a56e-4baf-b6a4-b09d5850459f   /127.0.0.1                     consumer-1
```

Are we OK with this change?